src/Graphics/PipelineCache.cs

```diff
@@ -681,9 +672,9 @@ public void EndApplySampler(SamplerStateCollection samplers, int register)
 
 		#region Private Helper Methods
 
-		private static unsafe int FloatToInt32(float f)
+		private static unsafe ulong FloatToUInt64(float f)
 		{
-			return *((int *) &f);
```

Comment:
This will probably have to stay as a uint conversion because of the type sizes, followed by a ulong cast at the end.

Aside from that I think all the signed<->unsigned conversions need the `unchecked` keyword to prevent exceptions (and maybe optimize things further?).

Reply:
Good catch, fixed.